CRM/Contribute/BAO/ContributionPage.php

```diff
@@ -45,6 +45,7 @@ public static function create($params) {
       CRM_Price_BAO_PriceFieldValue::updateFinancialType($params['id'], 'civicrm_contribution_page', $params['financial_type_id']);
     }
     CRM_Utils_Hook::post($hook, 'ContributionPage', $dao->id, $dao);
+    unset(\Civi::$statics['CRM_Core_PseudoConstant']);
```

Comment:
So that the values are refreshed after a new page is added

Reply:
I feel like this should really be done through a helper function of some kind rather than accessing the "raw" data directly. Effectively we want to clear that cache but we shouldn't need to know how the cache is internally stored.

Reply:
In fact, doesn't `CRM_Core_PseudoConstant::flush()` do what you need?

Reply:
Thanks @mattwire  - I've switched it over